test/ChainSpecificDeployment.js

```diff
@@ -0,0 +1,136 @@
+        expect(deployment.activeTokenDeployer.address).to.equal(deployment.interchainTokenDeployer.address);
+
+        // Verify that the standard deployer points to InterchainToken
+        const standardImplementation = await deployment.interchainTokenDeployer.implementationAddress();
+        expect(standardImplementation).to.equal(deployment.interchainToken.address);
+
+        // Test deploying a new token using the standard deployer
+        const salt = ethers.utils.randomBytes(32);
+        const tokenId = ethers.utils.randomBytes(32);
+        const tokenAddress = await deployment.activeTokenDeployer.deployedAddress(salt);
+
+        await deployment.activeTokenDeployer.deployInterchainToken(salt, tokenId, wallet.address, 'TestToken', 'TEST', 18);
+
+        // Verify the deployed token is a standard InterchainToken (no getDeployer function)
+        const token = await ethers.getContractAt('InterchainToken', tokenAddress, wallet);
+
+        // Standard InterchainToken should not have getDeployer function
+        expect(token.getDeployer).to.be.undefined;
+
+        console.log('✅ Standard deployment successful!');
+        console.log('- Deployed token address:', tokenAddress);
+        console.log('- Token type: Standard InterchainToken (no deployer function)');
+    });
+
+    it('should verify storage layout differences', async () => {
+        console.log('\n=== Testing Storage Layout Differences ===');
+
+        // Deploy both types
+        const hyperliquidDeployment = await deployAll(wallet, 'hyperliquid', [], 'HyperliquidTest', 'HyperliquidTestFactory');
+        const standardDeployment = await deployAll(wallet, 'avalanche', [], 'StandardTest', 'StandardTestFactory');
+
+        // Deploy tokens using each deployer
+        const salt1 = ethers.utils.randomBytes(32);
+        const salt2 = ethers.utils.randomBytes(32);
+        const tokenId = ethers.utils.randomBytes(32);
+
+        await hyperliquidDeployment.activeTokenDeployer.deployInterchainToken(salt1, tokenId, wallet.address, 'TestToken', 'TEST', 18);
+        await standardDeployment.activeTokenDeployer.deployInterchainToken(salt2, tokenId, wallet.address, 'TestToken', 'TEST', 18);
+
+        const hyperliquidTokenAddress = await hyperliquidDeployment.activeTokenDeployer.deployedAddress(salt1);
+        const standardTokenAddress = await standardDeployment.activeTokenDeployer.deployedAddress(salt2);
+
+        // Check slot 0 for both tokens
+        const provider = ethers.provider;
+        const hyperliquidSlot0 = await provider.getStorageAt(hyperliquidTokenAddress, 0);
+        const standardSlot0 = await provider.getStorageAt(standardTokenAddress, 0);
+
+        console.log('Storage slot 0 comparison:');
+        console.log('- Hyperliquid token slot 0:', hyperliquidSlot0);
+        console.log('- Standard token slot 0:', standardSlot0);
+
+        // Hyperliquid should have deployer in slot 0, standard should be empty
```

Comment:
we don't care what the standard token does since there's no requirement of what slot 0 there should or should not be. Just need to test the hyperliquid one for which there's a separate test already